ansq/tcp/types/client.py

```diff
@@ -0,0 +1,57 @@
+from typing import TYPE_CHECKING, Any, Dict, Mapping, Sequence, Tuple
+
+if TYPE_CHECKING:
+    from ansq.tcp.connection import NSQConnection
+
+
+class Client:
+    """Base class for reader and writer."""
+
+    def __init__(
+        self,
+        nsqd_tcp_addresses: Sequence[str],
+        connection_options: Mapping[str, Any] = None,
+    ):
+        self._nsqd_tcp_addresses = nsqd_tcp_addresses
+        self._connection_options = (
+            dict(connection_options) if connection_options is not None else {}
+        )
+        self._connections: Dict[str, NSQConnection] = {}
+
+        if not self._nsqd_tcp_addresses:
+            raise ValueError("nsqd_tcp_addresses must be specified")
+
+    async def connect(self) -> None:
+        """Connect to nsqd addresses."""
+        for address in self._nsqd_tcp_addresses:
+            try:
+                host, port = address.split(":")
+            except ValueError:
+                raise ValueError(f"Invalid TCP address: {address}")
+            await self.connect_to_nsqd(host=host, port=int(port))
+
+    async def close(self) -> None:
+        """Close all connections."""
+        for connection in self._connections.values():
+            await connection.close()
+
+    async def connect_to_nsqd(self, host: str, port: int) -> "NSQConnection":
+        """Connect and identify to nsqd by given host and port."""
+        from ansq.tcp.connection import NSQConnection
+
+        connection = NSQConnection(host=host, port=port, **self._connection_options)
```

Comment:
We know all the possible arguments `NSQConnection.__init__` could receive. They are all defined as keyword-only arguments. It looks like a good place to use a dataclass or an attrs class here. This will allow getting rid of `Mapping[str, Any]` for `connection_options`.

Reply:
This is a backward-incompatible change for sure. However, it looks like a helpful one.

Also, we could keep explicit arguments in `NSQConnection.__init__` that will take precedence if defined and will produce a deprecation warning.

Reply:
Could you elaborate on "dataclass or an attrs class" and how it would help to pass connection options from reader/writer? Any examples for `create_reader()`, `Reader()` and `NSQConnection()` would help.

Reply:
Will be improved in a separate PR